### PR TITLE
fd_shred: add metrics to tile and fd_fec_resolver

### PR DESCRIPTION
**Pull request by ptaffet-jump**
Add the following metric types:
```
# HELP shred_cluster_contact_info_cnt Number of contact infos in the cluster contact info message
# HELP shred_microblocks_abandoned The number of microblocks that were abandoned because we switched slots without finishing the current slot
# HELP shred_batch_sz The size (in bytes) of each microblock batch that is shredded
# HELP shred_batch_microblock_cnt The number of microblocks in each microblock batch that is shredded
# HELP shred_shredding_duration_seconds Duration of producing one FEC set from the shredder
# HELP shred_add_shred_duration_seconds Duration of verifying and processing one shred received from the network
# HELP shred_shred_processed_bad_slot The result of processing a thread from the network (Shred was for a slot for which we don't know the leader)
# HELP shred_shred_processed_parse_failed The result of processing a thread from the network (Shred parsing failed)
# HELP shred_shred_processed_rejected The result of processing a thread from the network (Shred was invalid for one of many reasons)
# HELP shred_shred_processed_ignored The result of processing a thread from the network (Shred was ignored because we had already received or reconstructed it)
# HELP shred_shred_processed_okay The result of processing a thread from the network (Shred accepted to an incomplete FEC set)
# HELP shred_shred_processed_completes The result of processing a thread from the network (Shred accepted and resulted in a valid, complete FEC set)
# HELP shred_fec_set_spilled The number of FEC sets that were spilled because they didn't complete in time and we needed space
# HELP shred_shred_rejected_initial The number shreds that were rejected before any resources were allocated for the FEC set
# HELP shred_fec_rejected_fatal The number of FEC sets that were rejected for reasons that cause the whole FEC set to become invalid
```

### CODE REVIEW

File: src/disco/shred/fd_fec_resolver.c [hunk 1]
```diff
@@ -402,6 +407,7 @@ int fd_fec_resolver_add_shred( fd_fec_resolver_t    * resolver,
       bmtrlist_push_head( bmtree_free_list, bmtree_mem );
       return FD_FEC_RESOLVER_SHRED_REJECTED;
     }
+    FD_MCNT_INC( SHRED, SHRED_REJECTED_INITIAL, (ulong)(-1L) ); /* Undo bad speculation */
```
Comment (mmcgee-jump) [inline src/disco/shred/fd_fec_resolver.c:410]: This is not valid, a counter cannot ever decrease in prometheus or the collector will assume that the application has restarted.

File: src/disco/shred/fd_fec_resolver.c [hunk 2]
```diff
@@ -568,6 +575,8 @@ int fd_fec_resolver_add_shred( fd_fec_resolver_t    * resolver,
     return FD_FEC_RESOLVER_SHRED_REJECTED;
   }
 
+  FD_MCNT_INC( SHRED, FEC_REJECTED_FATAL, (ulong)(-1L) ); /* It wasn't rejected, so undo the bad speculation */
```
Comment (mmcgee-jump) [inline src/disco/shred/fd_fec_resolver.c:578]: As above

File: src/app/fdctl/run/tiles/fd_metric.c [hunk 3]
```diff
@@ -263,6 +263,9 @@ prometheus_print( fd_topo_t * topo,
   PRINT( "\n" );
   result = prometheus_print1( topo, out, out_len, "store", FD_METRICS_STORE_TOTAL, FD_METRICS_STORE, PRINT_TILE );
   if( FD_UNLIKELY( result<0 ) ) return result;
+  PRINT( "\n" );
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_metric.c:266]: I have some notion to put `shred` before `store` in the output to preserve the ordering of the tiles in a fragment flow.

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
.

File: src/disco/shred/fd_fec_resolver.c (hunk 1 above)
Comment (ptaffet-jump) [reply src/disco/shred/fd_fec_resolver.c:410]: Ah, I forgot the metrics tile could actually observe the count in between these. I was thinking it required metric write, but that's not true.

File: src/app/fdctl/run/tiles/fd_metric.c (hunk 3 above)
Comment (ptaffet-jump) [reply src/app/fdctl/run/tiles/fd_metric.c:266]: Yeah, seems reasonable.

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**